nats.go

```diff
@@ -4273,15 +4312,19 @@ func (nc *Conn) SubscribeSync(subj string) (*Subscription, error) {
 		return nil, ErrInvalidConnection
 	}
 	mch := make(chan *Msg, nc.Opts.SubChanLen)
-	return nc.subscribe(subj, _EMPTY_, nil, mch, true, nil)
+	var errCh chan error
+	if nc.Opts.PermissionErrOnSubscribe {
+		errCh = make(chan error, 100)
```

Comment:
Good question, I set it to some value without thinking much about this and kind of forgot to analyze it properly. Since we really only push permission errors on this channel and we do it once per subscription, a buffered channel with size 1 should suffice (so that it's not blocking).